syntax = "proto2";

package ascend;

message heartbeat_msg {
  required double lng = 1;
  required double lat = 2;
  required double alt = 3;
  required int32 bat_percentage = 4;
}

message emergency_msg {
  required string name = 1;
}


message msg {
    oneof type {
        heartbeat_msg heartbeat = 1;
        emergency_msg emergency = 2;
    }   
}